cmd/kube-scheduler/app/server.go

```diff
@@ -179,6 +179,7 @@ func Run(cc schedulerserverconfig.CompletedConfig, stopCh <-chan struct{}) error
 		framework.NewRegistry(),
 		cc.ComponentConfig.Plugins,
 		cc.ComponentConfig.PluginConfig,
+		schedulerFeatureGateDependencies,
```

Comment:
Thinking more about it, why can't we have it as part of `schedulerserverconfig`?

Reply:
I tried this, but the versioned `KubeSchedulerConfig` type is hosted under `staging/kube-scheduler`, while we're adding `FeatureDependency` as a type in the `k/k/pkg/scheduler/api`. Currently `KubeSchedulerConfig` has no dependencies on that api and afaik `staging` can't depend on `k/k` right? So then this ends up breaking when you get to the internal/v1alpha1 conversions

Reply:
If we made `FeatureDependency` a type under the `KubeSchedulerConfig` config api instead of pkg/scheduler/api, I think that would work. Do you want to do that instead?